### PR TITLE
Fix graphicbuilder transform part 2

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
The previous fix to the graphicbuilder transform problem actually resulted in some graphicbuilder primitives being transformed twice because they already had the transform applied to them. This fixes that issue by moving the transformation only to primitives that were missing it. This also adds code to ensure that GeometryAccumulator does not mutate the graphicbuilder's transform in-place.

### CODE REVIEW

File: core/frontend/src/render/primitives/geometry/GeometryAccumulator.ts [hunk 1]
```diff
@@ -59,18 +59,20 @@ export class GeometryAccumulator {
     return range.isNull ? undefined : range;
   }
 
-  private calculateTransform(transform: Transform, range: Range3d): void {
-    if (this.haveTransform) this._transform.multiplyTransformTransform(transform, transform);
-    transform.multiplyRange(range, range);
+  private calculateTransform(transform: Transform, range: Range3d): Transform {
+    const xform = transform.clone();
```
Comment (pmconne) [inline core/frontend/src/render/primitives/geometry/GeometryAccumulator.ts:63]: Only clone if you have to mutate (`this.haveTransform`).

**Review by pmconne — CHANGES_REQUESTED** [review]

File: core/frontend/src/render/primitives/geometry/GeometryPrimitives.ts [hunk 2]
```diff
@@ -270,6 +260,9 @@ class SolidPrimitiveGeometry extends Geometry {
   protected _getPolyfaces(opts: StrokeOptions): PolyfacePrimitiveList {
     const builder = PolyfaceBuilder.create(opts);
     builder.addGeometryQuery(this._primitive);
-    return new PolyfacePrimitiveList(PolyfacePrimitive.create(this.displayParams, builder.claimPolyface()));
+    const pf = builder.claimPolyface();
+    if (!this.transform.isIdentity)
+      pf.tryTransformInPlace(this.transform);
```
Comment (pmconne) [inline core/frontend/src/render/primitives/geometry/GeometryPrimitives.ts:265]: Would be cheaper to transform the solid primitive before facetting than to transform the polyface.

**Comment by pmconne** [comment]
Did you determine whether the test you added in previous PR would have succeeded even without your code changes?
If so, why did it continue to succeed with your code changes?
Do you need a better test?

**Comment by markschlosseratbentley** [comment]
> Did you determine whether the test you added in previous PR would have succeeded even without your code changes?
> If so, why did it continue to succeed with your code changes?
> Do you need a better test?

@pmconne 
The test I added previously used an addShape decoration.  I incorrectly assumed that addShape decorations were not transforming properly like solidPrimitives.  Since the code change resulted in the transform being incorrectly applied twice to addShape decorations, the test "correctly" detected that the transformed decoration moved on the screen differently than before the code change.  My assumption of the problem was faulty.

I have added a test which actually tests the transform problem with a solidPrimitive.  This test correctly fails before this code change and succeeds after this code change.

File: core/frontend/src/render/primitives/geometry/GeometryAccumulator.ts [hunk 3]
```diff
@@ -60,7 +60,7 @@ export class GeometryAccumulator {
   }
 
   private calculateTransform(transform: Transform, range: Range3d): Transform {
-    const xform = transform.clone();
+    const xform = this.haveTransform ? transform : transform.clone();
```
Comment (pmconne) [inline core/frontend/src/render/primitives/geometry/GeometryAccumulator.ts:63]: Sorry to nitpick but
```
private calculateTransform(transform: Transform, range: Range3d): Transform {
  if (this.haveTransform)
    transform = this._transform.multiplyTransformTransform(transform);

  transform.multiplyRange(range, range);
  return transform;
}
```

Comment (markschlosseratbentley) [reply core/frontend/src/render/primitives/geometry/GeometryAccumulator.ts]: All good. That's a better usage of the API; thanks.

**Review by pmconne — APPROVED** [review]

**Pull request merged**